sdk/eventhub/event-hubs/test/eventProcessor.spec.ts

```diff
@@ -735,6 +749,95 @@ describe("Event Processor", function(): void {
       partitionOwnershipMap.get(processorByName[`processor-0`].id)!.length.should.oneOf([n, n + 1]);
       partitionOwnershipMap.get(processorByName[`processor-1`].id)!.length.should.oneOf([n, n + 1]);
     });
+
+    it("should ensure that all the processors maintain a steady-state when all partitions are being processed", async function(): Promise<
+      void
+    > {
+      const partitionIds = await client.getPartitionIds({});
+      const checkpointStore = new InMemoryCheckpointStore();
+      const claimedPartitionsSet = new Set<string>();
+
+      let allPartitionsClaimed = false;
+      let thrashAfterSettling = false;
+      const handlers: SubscriptionEventHandlers = {
+        async processInitialize(context) {
+          // For this test we don't want to actually checkpoint, just test ownership.
+          context.setStartPosition(EventPosition.latest());
+          if (allPartitionsClaimed) {
+            thrashAfterSettling = true;
+            return;
+          }
+          claimedPartitionsSet.add(context.partitionId);
```

Comment:
Do you also want to remove the partition on processClose() if the partition is stolen?

Reply:
If it's stolen then the other processor already claimed it (and added it to the set).

Reply:
Yeah, I was trying to think if it would be possible to think that all partitions are fully claimed (and steady) when things are still thrashing.

Reply:
Made updates to the test to be more resilient.